In [1]:
from openai import OpenAI

client = OpenAI(api_key='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')

import datasets

dataset=datasets.load_dataset("PiC/phrase_similarity")

/home/ansh/.local/lib/python3.10/site-packages/tqdm/auto.py:21: TqdmWarning: IProgress not found. Please update jupyter and ipywidgets. See https://ipywidgets.readthedocs.io/en/stable/user_install.html
  from .autonotebook import tqdm as notebook_tqdm


In [2]:
sentences1=[dataset['test'][i]['sentence1'] for i in range(len(dataset['test']))]
sentences2=[dataset['test'][i]['sentence2'] for i in range(len(dataset['test']))]
phrase1=[dataset['test'][i]['phrase1'] for i in range(len(dataset['test']))]
phrase2=[dataset['test'][i]['phrase2'] for i in range(len(dataset['test']))]
labels=[str(dataset['test'][i]['label']) for i in range(len(dataset['test']))]
print((labels[0]))

print(phrase1[0])
print(phrase2[0])

0
air position
posture while jumping


In [3]:
messages = [
    {"role": "system", "content": "You are a kind helpful assistant."},
]
messages.append(
    {"role": "user", "content": "Reply only in 0 or 1 for the following questions. DONOT USE ANY OTHER WORDS. ONLY ONE CHARATER REPLIES ARE ALLOWED. "}
)

chat_completion = client.chat.completions.create(
    messages=messages,
    model="gpt-3.5-turbo",
)
cont = chat_completion.choices[0].message.content
print(cont)
messages.append( {"role": "assistant", "content": cont})

Sure.


In [4]:
import time
time.sleep(20)
correct=0
times=100

outputList=[]

for i in range(times):
    messages.append({"role": "user", "content": "Do these two phrases mean the same thing?\n\n"+phrase1[i]+'\n'+phrase2[i]+"\n\n To avoid ambiguity the phrases are used in these two sentences:\n\n"+sentences1[i]+"\n"+sentences2[i]+"\n JUDGE BY OBSERVING THESE SENTENCES CAREFULLY NOT ONLY PHRASES AND TELL IF THE TWO PHRASES HAVE THE SAME MEANING"})
    chat_completion = client.chat.completions.create(
        messages=messages,
        model="gpt-3.5-turbo",
    )
    cont = chat_completion.choices[0].message.content
    print('"',phrase1[i],'"')
    print('"',phrase2[i],'"')
    print("GPT: ",cont, "expected: ",labels[i])

    if cont==labels[i]:
        correct+=1
    elif labels[i] in cont:
        correct+=1
    print("accuracy: ",correct/(i+1))
    print()
    outputList.append([sentences1[i], sentences2[i], cont, labels[i], correct/(i+1)])
    if times!=i-1:
        time.sleep(20)
    messages.pop()

" air position "
" posture while jumping "
GPT:  0 expected:  0
accuracy:  1.0

" associated track "
" correlating music single "
GPT:  0 expected:  0
accuracy:  1.0

" whole parts "
" extended sections "
GPT:  0 expected:  1
accuracy:  0.6666666666666666

" wide set "
" spacious collection "
GPT:  0 expected:  0
accuracy:  0.75

" full protection "
" complete defense "
GPT:  0 expected:  0
accuracy:  0.8

" prior case "
" preceding game "
GPT:  0 expected:  0
accuracy:  0.8333333333333334

" another station "
" a separate airport "
GPT:  0 expected:  0
accuracy:  0.8571428571428571

" initial activity "
" starting task "
GPT:  0 expected:  1
accuracy:  0.75

" single square "
" solitary border "
GPT:  0 expected:  1
accuracy:  0.6666666666666666

" independent operation "
" individual enterprise "
GPT:  0 expected:  0
accuracy:  0.7

" long segment "
" quite a stretch "
GPT:  0 expected:  0
accuracy:  0.7272727272727273

" borg family "
" Borg household "
GPT:  0 expected:  0
accuracy



## 5 queries

air position
posture while jumping
GPT:  0 expected:  0
accuracy:  1.0

associated track
correlating music single
GPT:  0 expected:  0
accuracy:  1.0

whole parts
extended sections
GPT:  0 (Yes) expected:  1
accuracy:  0.6666666666666666

wide set
spacious collection
GPT:  0 expected:  0
accuracy:  0.75

full protection
complete defense
GPT:  0 expected:  0
accuracy:  0.8

Final accuracy: 100% (3rd one it replied 1(YES) instead of 1 leading to accuracy getting read incorrectly)

## 100 Queries

" air position "
" posture while jumping "
GPT:  0 expected:  0
accuracy:  1.0

" associated track "
" correlating music single "
GPT:  0 expected:  0
accuracy:  1.0

" whole parts "
" extended sections "
GPT:  0 expected:  1
accuracy:  0.6666666666666666

" wide set "
" spacious collection "
GPT:  0 expected:  0
accuracy:  0.75

" full protection "
" complete defense "
GPT:  0 expected:  0
accuracy:  0.8

" prior case "
" preceding game "
GPT:  0 expected:  0
accuracy:  0.8333333333333334

" another station "
" a separate airport "
GPT:  0 expected:  0
accuracy:  0.8571428571428571

" initial activity "
" starting task "
GPT:  0 expected:  1
accuracy:  0.75

" single square "
" solitary border "
GPT:  1 expected:  1
accuracy:  0.7777777777777778

" independent operation "
" individual enterprise "
GPT:  0 expected:  0
accuracy:  0.8

" long segment "
" quite a stretch "
GPT:  0 expected:  0
accuracy:  0.8181818181818182

" borg family "
" Borg household "
GPT:  0 expected:  0
accuracy:  0.8333333333333334

" one die "
" a single counter "
GPT:  0 expected:  0
accuracy:  0.8461538461538461

" material one "
" physical world "
GPT:  0 expected:  0
accuracy:  0.8571428571428571

" offensive element "
" attack component "
GPT:  0 expected:  0
accuracy:  0.8666666666666667

" fast movements "
" quick motions "
GPT:  1 expected:  1
accuracy:  0.875

" unnecessary credit "
" gratuitous acknowledgment "
GPT:  0 expected:  1
accuracy:  0.8235294117647058

" group address "
" overall location "
GPT:  0 expected:  0
accuracy:  0.8333333333333334

" third incarnation "
" third embodiment "
GPT:  0 expected:  0
accuracy:  0.8421052631578947

" visual record "
" perceivable history "
GPT:  0 expected:  0
accuracy:  0.85

" great dash "
" awesome speed "
GPT:  0 expected:  0
accuracy:  0.8571428571428571

" danger zone "
" hazard range "
GPT:  1 expected:  1
accuracy:  0.8636363636363636

" first address "
" inaugural speech "
GPT:  0 expected:  0
accuracy:  0.8695652173913043

" mobile frame "
" flexible abstract coordinate system "
GPT:  1 expected:  1
accuracy:  0.875

" interchangeable part "
" exchangeable segment "
GPT:  0 expected:  1
accuracy:  0.84

" new bit "
" fresh segment "
GPT:  0 expected:  1
accuracy:  0.8076923076923077

" other person "
" anyone else "
GPT:  0 expected:  0
accuracy:  0.8148148148148148

" sole action "
" particular statement "
GPT:  1 expected:  1
accuracy:  0.8214285714285714

" fledgling system "
" novel approach "
GPT:  0 expected:  0
accuracy:  0.8275862068965517

" battery fire "
" combustion of its power supply "
GPT:  0 expected:  1
accuracy:  0.8

" particularly difficult subject "
" specifically complex topic "
GPT:  0 expected:  1
accuracy:  0.7741935483870968

" primary face "
" public figure "
GPT:  0 expected:  1
accuracy:  0.75

" later impact "
" future influence "
GPT:  0 expected:  0
accuracy:  0.7575757575757576

" media manipulation "
" controlling the news "
GPT:  0 expected:  1
accuracy:  0.7352941176470589

" second casting "
" subsequent mold "
GPT:  0 expected:  0
accuracy:  0.7428571428571429

" first suggestion "
" starting advice "
GPT:  0 expected:  0
accuracy:  0.75

" loose end "
" point of unsureness "
GPT:  0 expected:  0
accuracy:  0.7567567567567568

" larger craft "
" bigger vehicle "
GPT:  0 expected:  1
accuracy:  0.7368421052631579

" whole scene "
" full landscape "
GPT:  0 expected:  1
accuracy:  0.717948717948718

" another arrangement "
" a different order "
GPT:  0 expected:  1
accuracy:  0.7

" chosen course "
" preferred fairway "
GPT:  0 expected:  1
accuracy:  0.6829268292682927

" middle ground "
" point of compromise "
GPT:  0 expected:  0
accuracy:  0.6904761904761905

" white writing "
" ivory letters "
GPT:  0 expected:  0
accuracy:  0.6976744186046512

" fair way "
" decent distance "
GPT:  0 expected:  0
accuracy:  0.7045454545454546

" major stature "
" considerable standing "
GPT:  1 expected:  1
accuracy:  0.7111111111111111

" least one cylinder "
" single or additional phonograph instrument "
GPT:  0 expected:  0
accuracy:  0.717391304347826

" inferior division "
" lesser section "
GPT:  0 expected:  0
accuracy:  0.723404255319149

" one individual "
" one figure "
GPT:  0 expected:  1
accuracy:  0.7083333333333334

" white light "
" quantitative phase imaging method "
GPT:  0 expected:  1
accuracy:  0.6938775510204082

" middle ground "
" happy medium "
GPT:  No expected:  1
accuracy:  0.68

" tough course "
" difficult time "
GPT:  No. expected:  1
accuracy:  0.6666666666666666

" basic background "
" simple history "
GPT:  0 expected:  1
accuracy:  0.6538461538461539

" color map "
" painted chart "
GPT:  0 expected:  1
accuracy:  0.6415094339622641

" following picture "
" next image "
GPT:  0 expected:  0
accuracy:  0.6481481481481481

" ideal size "
" prime amount "
GPT:  0 expected:  0
accuracy:  0.6545454545454545

" whose word "
" their meaning "
GPT:  0 expected:  0
accuracy:  0.6607142857142857

" major degrees "
" considerable severity "
GPT:  0 expected:  1
accuracy:  0.6491228070175439

" robust selection "
" powerful choice "
GPT:  0 expected:  0
accuracy:  0.6551724137931034

" permit process "
" allowed court document "
GPT:  No expected:  0
accuracy:  0.6440677966101694

" human response "
" the impact by the population "
GPT:  0 expected:  1
accuracy:  0.6333333333333333

" star figure "
" renown leader "
GPT:  1 expected:  1
accuracy:  0.639344262295082

" organic structure "
" natural composition "
GPT:  0 expected:  1
accuracy:  0.6290322580645161

" conduct rules "
" behavior regulations "
GPT:  Yes (1). expected:  1
accuracy:  0.6349206349206349

" local terms "
" simple words "
GPT:  1 expected:  0
accuracy:  0.625

" ancient use "
" classical Latin rite "
GPT:  1 expected:  1
accuracy:  0.6307692307692307

" exposure limits "
" financial risk restrictions "
GPT:  0 expected:  1
accuracy:  0.6212121212121212

" local peaks "
" nearby mountains "
GPT:  1 expected:  0
accuracy:  0.6119402985074627

" high evaluation "
" tall estimation "
GPT:  0 expected:  0
accuracy:  0.6176470588235294

" little relationship "
" minimal connection "
GPT:  0 expected:  1
accuracy:  0.6086956521739131

" true mark "
" real sign "
GPT:  0 expected:  1
accuracy:  0.6

" separate seat "
" different chair "
GPT:  No (0). expected:  0
accuracy:  0.6056338028169014

" integrated access "
" computer premises device "
GPT:  0 expected:  0
accuracy:  0.6111111111111112

" click models "
" a fashion company "
GPT:  0 expected:  0
accuracy:  0.6164383561643836

" little news "
" some disclosures "
GPT:  0 expected:  1
accuracy:  0.6081081081081081

" key tracks "
" major railroads "
GPT:  0 expected:  0
accuracy:  0.6133333333333333

" little return "
" meager rebound "
GPT:  0 expected:  0
accuracy:  0.618421052631579

" commercial material "
" Marketable substance "
GPT:  0 expected:  0
accuracy:  0.6233766233766234

" real meat "
" useful discussions "
GPT:  0 expected:  1
accuracy:  0.6153846153846154

" survey methods "
" polling procedures "
GPT:  0 expected:  0
accuracy:  0.620253164556962

" contact centre "
" meeting place for nonresident parents and children "
GPT:  0 expected:  0
accuracy:  0.625

" largest difference "
" biggest contrast "
GPT:  0 expected:  1
accuracy:  0.6172839506172839

" dull game "
" tedious pastime "
GPT:  0 expected:  0
accuracy:  0.6219512195121951

" well-organized system "
" structured arrangement "
GPT:  0 expected:  0
accuracy:  0.6265060240963856

" safe ground "
" inoffensive conversation landscape "
GPT:  1 expected:  0
accuracy:  0.6190476190476191

" second duty "
" additional responsibility "
GPT:  1 expected:  1
accuracy:  0.6235294117647059

" subordinate role "
" secondary part "
GPT:  1 expected:  0
accuracy:  0.6162790697674418

" driving school "
" "Driving School" "
GPT:  0 expected:  0
accuracy:  0.6206896551724138

" excessive roughness "
" exorbitant bumpiness "
GPT:  0 expected:  0
accuracy:  0.625

" relevant authority "
" applicable leadership "
GPT:  0 expected:  0
accuracy:  0.6292134831460674

" early interception "
" advanced prevention "
GPT:  0 expected:  1
accuracy:  0.6222222222222222

" quiet design "
" plain blueprint "
GPT:  1 expected:  1
accuracy:  0.6263736263736264

" 1st cycle "
" first round "
GPT:  0 expected:  0
accuracy:  0.6304347826086957

" transitional season "
" turning-point interval "
GPT:  1 expected:  1
accuracy:  0.6344086021505376

" last hours "
" Last Hours "
GPT:  1 expected:  1
accuracy:  0.6382978723404256

" direct derivative "
" straight outgrowth "
GPT:  0 expected:  0
accuracy:  0.6421052631578947

" significant thrust "
" a lot of power "
GPT:  1 expected:  1
accuracy:  0.6458333333333334

" first score "
" beginning point "
GPT:  1 expected:  0
accuracy:  0.6391752577319587

" small corner "
" minor area "
GPT:  1 expected:  0
accuracy:  0.6326530612244898

" common support "
" regular tool "
GPT:  0 expected:  1
accuracy:  0.6262626262626263

" complete answer "
" entire response "
GPT:  1 expected:  0
accuracy:  0.62

# Accuracy: 62% 